--- a/chrome/browser/importer/importer_list.cc
+++ b/chrome/browser/importer/importer_list.cc
@@ -254,9 +254,4 @@ std::vector<user_data_importer::SourcePr
     profiles.push_back(bookmarks_profile);
   }
-  for (auto& profile : profiles) {
-    profile.services_supported &= ~(user_data_importer::PASSWORDS
-                                    | user_data_importer::AUTOFILL_FORM_DATA);
-  }
-
   return profiles;
 }
--- a/components/password_manager/core/browser/password_suggestion_generator.cc
+++ b/components/password_manager/core/browser/password_suggestion_generator.cc
@@ -397,9 +397,5 @@ PasswordSuggestionGenerator::PasswordSug
 void PasswordSuggestionGenerator::AppendOptionalFooterSection(
     std::vector<autofill::Suggestion>* suggestions) const {
-  if ((true)) {
-    return;
-  }
-
   bool has_webauthn_credential = std::ranges::any_of(
       *suggestions,
       [](SuggestionType type) {
--- a/chrome/browser/ui/omnibox/omnibox_pedal_implementations.cc
+++ b/chrome/browser/ui/omnibox/omnibox_pedal_implementations.cc
@@ -2028,7 +2028,12 @@ GetPedalImplementations(bool incognito,
   if (!incognito && !guest) {
     add(new OmniboxPedalClearBrowsingData(/*incognito=*/false));
   }
+  add(new OmniboxPedalManagePasswords());
+  add(new OmniboxPedalUpdateCreditCard());
   add(new OmniboxPedalLaunchIncognito());
+  if (!base::android::device_info::is_automotive()) {
+    add(new OmniboxPedalRunChromeSafetyCheck());
+  }
   add(new OmniboxPedalPlayChromeDinoGame());
   add(new OmniboxPedalManageSiteSettings());
   add(new OmniboxPedalManageChromeSettings());
@@ -2040,10 +2045,14 @@ GetPedalImplementations(bool incognito,
   if (!guest) {
     add(new OmniboxPedalClearBrowsingData(incognito));
   }
+  add(new OmniboxPedalManagePasswords());
+  add(new OmniboxPedalUpdateCreditCard());
   add(new OmniboxPedalLaunchIncognito());
   add(new OmniboxPedalUpdateChrome());
   add(new OmniboxPedalManageSecuritySettings());
   add(new OmniboxPedalManageCookies());
+  add(new OmniboxPedalManageAddresses());
+  add(new OmniboxPedalManageSync());
   add(new OmniboxPedalManageSiteSettings());
   if (incognito) {
     add(new OmniboxPedalCloseIncognitoWindows());
--- a/chrome/browser/ui/views/location_bar/location_bar_view.cc
+++ b/chrome/browser/ui/views/location_bar/location_bar_view.cc
@@ -507,9 +507,11 @@ void LocationBarView::Init() {
     if (optimization_guide::features::ShouldEnableOptimizationGuideIconView()) {
       params.types_enabled.push_back(PageActionIconType::kOptimizationGuide);
     }
     params.types_enabled.push_back(PageActionIconType::kClickToCall);
+    params.types_enabled.push_back(PageActionIconType::kAutofillAddress);
+    params.types_enabled.push_back(PageActionIconType::kManagePasswords);
     if (!apps::features::ShouldShowLinkCapturingUX()) {
       params.types_enabled.push_back(PageActionIconType::kIntentPicker);
     }
     params.types_enabled.push_back(PageActionIconType::kBookmarkStar);
     params.types_enabled.push_back(PageActionIconType::kTranslate);
@@ -520,6 +522,12 @@ void LocationBarView::Init() {
     params.types_enabled.push_back(PageActionIconType::kZoom);
     params.types_enabled.push_back(PageActionIconType::kFileSystemAccess);
+    params.types_enabled.push_back(PageActionIconType::kCookieControls);
+    params.types_enabled.push_back(
+        PageActionIconType::kPaymentsOfferNotification);
     params.types_enabled.push_back(PageActionIconType::kMemorySaver);
   }
+  params.types_enabled.push_back(PageActionIconType::kSaveCard);
+  params.types_enabled.push_back(PageActionIconType::kSaveIban);
+  params.types_enabled.push_back(PageActionIconType::kFilledCardInformation);
   params.types_enabled.push_back(PageActionIconType::kLocalCardMigration);
   params.types_enabled.push_back(PageActionIconType::kOffersAndRewards);
--- a/chrome/browser/ui/browser_actions.cc
+++ b/chrome/browser/ui/browser_actions.cc
@@ -808,5 +808,104 @@ void BrowserActions::InitializeBrowserAc
           .SetVisible(!sharing_hub::SharingIsDisabledByPolicy(profile))
           .Build());
+  root_action_item_->AddChild(
+      ChromeMenuAction(
+          base::BindRepeating(
+              [](BrowserWindowInterface* bwi, TabStripModel* tab_strip_model,
+                 actions::ActionItem* item,
+                 actions::ActionInvocationContext context) {
+                auto page_action_trigger =
+                    context.GetProperty(page_actions::kPageActionTriggerKey);
+                // If triggered by omnibox page action, do nothing.
+                if (page_action_trigger !=
+                    page_actions::kInvalidPageActionTrigger) {
+                  return;
+                }
+
+                auto* controller = autofill::AddressBubblesIconController::Get(
+                    tab_strip_model->GetActiveWebContents());
+                if (controller && controller->GetBubbleView()) {
+                  controller->GetBubbleView()->Hide();
+                } else {
+                  chrome::ShowAddresses(bwi);
+                }
+              },
+              bwi, tab_strip_model),
+          kActionShowAddressesBubbleOrPage,
+          IDS_ADDRESSES_AND_MORE_SUBMENU_OPTION,
+          IDS_ADDRESSES_AND_MORE_SUBMENU_OPTION,
+          vector_icons::kLocationOnChromeRefreshIcon)
+          .SetEnabled(!is_guest_session)
+          .Build());
+
+  root_action_item_->AddChild(
+      ChromeMenuAction(
+          base::BindRepeating(
+              [](BrowserWindowInterface* bwi, TabStripModel* tab_strip_model,
+                 actions::ActionItem* item,
+                 actions::ActionInvocationContext context) {
+                auto page_action_trigger =
+                    context.GetProperty(page_actions::kPageActionTriggerKey);
+                // When page action is migrated, clicking on the omnibox page
+                // should not close the bubble or navigate to `Payment Methods`
+                // settings page.
+                // Page action trigger is a valid value only when this action
+                // is triggered from the migrated page action icon.
+                if (page_action_trigger !=
+                    page_actions::kInvalidPageActionTrigger) {
+                  return;
+                }
+
+                auto hide_bubble = [tab_strip_model](int command_id) -> bool {
+                  auto* controller = autofill::SavePaymentIconController::Get(
+                      tab_strip_model->GetActiveWebContents(), command_id);
+                  if (controller && controller->GetPaymentBubbleView()) {
+                    controller->GetPaymentBubbleView()->Hide();
+                    return true;
+                  }
+                  return false;
+                };
+                const bool bubble_hidden =
+                    hide_bubble(IDC_SAVE_CREDIT_CARD_FOR_PAGE) ||
+                    hide_bubble(IDC_SAVE_IBAN_FOR_PAGE);
+                if (!bubble_hidden) {
+                  chrome::ShowPaymentMethods(bwi);
+                }
+              },
+              bwi, tab_strip_model),
+          kActionShowPaymentsBubbleOrPage, IDS_PAYMENT_METHOD_SUBMENU_OPTION,
+          IDS_PAYMENT_METHOD_SUBMENU_OPTION, kCreditCardChromeRefreshIcon)
+          .SetEnabled(!is_guest_session)
+          .Build());
+
+  root_action_item_->AddChild(
+      ChromeMenuAction(
+          base::BindRepeating(
+              [](BrowserWindowInterface* bwi, TabStripModel* tab_strip_model,
+                 actions::ActionItem* item,
+                 actions::ActionInvocationContext context) {
+                content::WebContents* const web_contents =
+                    tab_strip_model->GetActiveWebContents();
+                if (PasswordsModelDelegateFromWebContents(web_contents)
+                        ->GetState() == password_manager::ui::INACTIVE_STATE) {
+                  chrome::ShowPasswordManager(bwi);
+                } else {
+                  auto* const controller =
+                      ManagePasswordsUIController::FromWebContents(
+                          web_contents);
+                  if (controller->IsShowingBubble()) {
+                    controller->HideBubble(
+                        /*initiated_by_bubble_manager=*/false);
+                  } else {
+                    chrome::ManagePasswordsForPage(bwi);
+                  }
+                }
+              },
+              bwi, tab_strip_model),
+          kActionShowPasswordsBubbleOrPage, IDS_VIEW_PASSWORDS,
+          IDS_VIEW_PASSWORDS, vector_icons::kPasswordManagerIcon)
+          .SetEnabled(!is_guest_session)
+          .Build());
+
   root_action_item_->AddChild(
       actions::ActionItem::Builder(
           base::BindRepeating(
--- a/chrome/browser/ui/web_applications/app_browser_controller.cc
+++ b/chrome/browser/ui/web_applications/app_browser_controller.cc
@@ -409,5 +409,6 @@ std::vector<actions::ActionId> AppBrowse
   std::vector<actions::ActionId> types_enabled = {
       kActionFind,
+      kActionShowPasswordsBubbleOrPage,
       kActionZoomNormal,
       kActionShowFileSystemAccess,
       kActionShowCookieControls,
@@ -436 +436,2 @@ AppBrowserController::GetTitleBarPageAct
  types_enabled.push_back(PageActionIconType::kFind);
+  types_enabled.push_back(PageActionIconType::kManagePasswords);
@@ -438 +439,3 @@ AppBrowserController::GetTitleBarPageAct
  types_enabled.push_back(PageActionIconType::kFileSystemAccess);
+  types_enabled.push_back(PageActionIconType::kSaveCard);
+  types_enabled.push_back(PageActionIconType::kAutofillAddress);
--- a/chrome/browser/ui/webui/side_panel/customize_chrome/customize_toolbar/customize_toolbar_handler.cc
+++ b/chrome/browser/ui/webui/side_panel/customize_chrome/customize_toolbar/customize_toolbar_handler.cc
@@ -56,6 +56,13 @@ MojoActionForChromeAction(actions::Actio
       return side_panel::customize_chrome::mojom::ActionId::kForward;
     case kActionNewIncognitoWindow:
       return side_panel::customize_chrome::mojom::ActionId::kNewIncognitoWindow;
+    case kActionShowPasswordsBubbleOrPage:
+      return side_panel::customize_chrome::mojom::ActionId::
+          kShowPasswordManager;
+    case kActionShowPaymentsBubbleOrPage:
+      return side_panel::customize_chrome::mojom::ActionId::kShowPaymentMethods;
+    case kActionShowAddressesBubbleOrPage:
+      return side_panel::customize_chrome::mojom::ActionId::kShowAddresses;
     case kActionShowDownloads:
       return side_panel::customize_chrome::mojom::ActionId::kShowDownloads;
     case kActionClearBrowsingData:
@@ -90,6 +97,12 @@ std::optional<actions::ActionId> ChromeA
       return kActionForward;
     case side_panel::customize_chrome::mojom::ActionId::kNewIncognitoWindow:
       return kActionNewIncognitoWindow;
+    case side_panel::customize_chrome::mojom::ActionId::kShowPasswordManager:
+      return kActionShowPasswordsBubbleOrPage;
+    case side_panel::customize_chrome::mojom::ActionId::kShowPaymentMethods:
+      return kActionShowPaymentsBubbleOrPage;
+    case side_panel::customize_chrome::mojom::ActionId::kShowAddresses:
+      return kActionShowAddressesBubbleOrPage;
     case side_panel::customize_chrome::mojom::ActionId::kShowDownloads:
       return kActionShowDownloads;
     case side_panel::customize_chrome::mojom::ActionId::kClearBrowsingData:
--- a/chrome/browser/ui/toolbar/app_menu_model.cc
+++ b/chrome/browser/ui/toolbar/app_menu_model.cc
@@ -1844 +1844,13 @@ void AppMenuModel::Build() {
+  if (!browser_->profile()->IsGuestSession()) {
+    sub_menus_.push_back(
+        std::make_unique<PasswordsAndAutofillSubMenuModel>(this));
+    AddSubMenuWithStringIdAndVectorIcon(
+        this, IDC_PASSWORDS_AND_AUTOFILL_MENU,
+        IDS_PASSWORDS_AND_AUTOFILL_MENU, sub_menus_.back().get(),
+        vector_icons::kPasswordManagerIcon);
+    SetElementIdentifierAt(
+        GetIndexOfCommandId(IDC_PASSWORDS_AND_AUTOFILL_MENU).value(),
+        kPasswordAndAutofillMenuItem);
+  }
+
  if (!browser_->profile()->IsOffTheRecord()) {
